### DISCUSSION TITLE
H * M ?

### DISCUSSION THREAD

**Discussion by ElmarBorgmeier** [Green-Software-Foundation/sci#228] [SCI feedback]
Hi,

I’m Elmar from the Green-Software-Design-R&D-Team at Syngenio (www.greensoftwaredesign.com – in german. An english version will follow soon). We developed an approach to rate the „greenness“ of software. Since we look at the issue from a slightly different angle, our rating structure differs from the SCI. But the SCI might fit nicely into our concept. 

It is great that you address the topic! It is indeed useful for development teams and product owners to get a grip on the climate-friendliness of their software. A scoring or rating approach will give them the support they need. 

The SCI is similar to the emission intensity G305-4 as defined by the GRI. SCI is basically a variant focused on a specific software system instead of an organization. Such a link to established reporting standards is a good thing.

The formula is: SCI = ((E*I) + M) per R

E is defined as: “Energy consumed by software in kWh”. Strictly speaking, software as such does not consume energy. It only makes hardware do it. More precise wording might avoid confusion here.

Both E and M depend on the hardware. This makes the SCI a score of a “software system”, meaning: it scores a specific combination of software and hardware. It is not a “software” score as such. The web-post mostly describes that correctly. It just could be more explicit about it.

E and M cover the two types of impact that software (indirectly) has on emissions. I do not yet understand why they are treated differently in the equation. “E*I” makes the conversion rate from energy to emissions explicit. M , however, includes this conversion implicitly. I suggest to replace M with “H*M”. In this notation, H refers to the amount of hardware used by the software, and M is the conversion rate from Hardware to emissions.

The introduction of H would make it easier for development teams to calculate the SCI. "E" can be measured, “I” is fix within a given environment. The team can then use a reference hardware H for load+performance testing, which in turn provides the data needed to calculate R. (R being something like “number of user sessions that can be handled by H”). M is a list of fixed values for the different types of hardware. 

This way, a team can measure the SCI of their reference software system and monitor its performance over time. And this is very useful indeed.

Can the SCI also be used to compare one software product to another? Differences in the reference hardware will lead to different values. Differences in the usage scenarios used to measure E will also result in different SCIs. The inclusion of “I” may actually lead to strange effects: The SCI becomes dependent on the cloud computing center used, because “I” will be different for the CCs. But the SCI is meant to be a software benchmark, not to rate the efficiency of CCs. 

Again, thank you for the initiative. We really need such a benchmark for the climate-friendliness of software.

**Comment by ElmarBorgmeier** [comment]
I guess I figured out myself why there is no "H" in the formula: To stress the fact that this is a green software score, not a hardware-based-green-IT thingy.

Creating an optical illusion of hardware absence is not the solution. Let's face it: Software itself does not have emissions. It always the hardware.
The only thing we can do to literally "take the hardware out of the equation" : make it a constant rather than a variable.

**Comment by atg-abhishek** [comment]
Thank you for sharing your thoughts @ElmarBorgmeier

**Comment by Henry-WattTime** [comment]
Working Group: @ElmarBorgmeier Would you be willing to join a conversation about this with the working group?

**Reply by ElmarBorgmeier** [reply]
I'd be happy to!

**Reply by atg-abhishek** [reply]
Great @ElmarBorgmeier !

**Reply by ElmarBorgmeier** [reply]
I'm happy to announce that my employer, Syngenio AG, joined GSF. (The Logo's already on the website.) And I joint the Standards Working Group. So: See you on thursday. :)

**Reply by atg-abhishek** [reply]
Welcome to the GSF @ElmarBorgmeier - looking forward to speaking with you in a few hours :)

**Comment by atg-abhishek** [comment]
@ElmarBorgmeier when you get a moment, could you drop in the reference for the G305-4 from the GRI? Thanks!

**Reply by ElmarBorgmeier** [reply]
It's on page 13 of this document: https://www.globalreporting.org/standards/media/1012/gri-305-emissions-2016.pdf

**Reply by atg-abhishek** [reply]
Thanks!
